### ISSUE TITLE
WordPressPCL 1.3.3 to 1.4.4 and WordPress 4.9.6

### ISSUE THREAD

**Issue by sinmax01** [wp-net/WordPressPCL#123]
I think there is a bug in WordPress 4.9.6. Or is there an incompatibility between these versions?

I get random errors on RequestJWToken, CustomRequest.Create, CustomRequest.Update. Let's say 10 requests work, then the next one crashes. At the next run, 17 requests work and the next one crashes. The requests seem to hang/timeout.

Sometimes the RequestJWToken does not wait or get a token, although the .Wait() method is called.

I don't see the requests in WordPress REST API Log plugin.

It was working perfect on a previous version of WordPress.

**Comment by ThomasPe** [comment]
I'm currently not seeing any of the issues you've described. Have you checked your server-logs to see if there's maybe an issue with traffic?

I'm also not seing any big changes being made in 4.9.6 to the REST API that would explain your symptoms. 

What plugin are you using the CustomRequest-Calls for? If you want you can add tests for these to the tests project for me to run them.

**Comment by sinmax01** [comment]
I am using CustomRequests for Advanced Custom Fields plugin in WordPress and for usual posts. It crashes at both usual Posts and Advanced Custom Fields. It seems to be related to the httpHelper async code.
When using the endpoint for posts (client.CustomRequest.Update<Post, Post>("wp/v2/my_post_type", myObject).Result), the exception is: "A task was canceled." (Version 1.3.3. returned null and no error was thrown).
When using the endpoint for ACF (client.CustomRequest.Create<MyACF, MyACF>("acf/v3/pages/" + postID, myACF).Wait(30 * 1000)), it times out after 30 seconds.
I installed the REST API Log in WordPress and the request where it crashes are not found in the log.

**Comment by ThomasPe** [comment]
I'll try to set up some tests using the CustomRequest method with ACF. TBH I haven't used this feature much myself yet.

**Comment by sinmax01** [comment]
It seems the issue was related to HTTP traffic. I still don't believe it yet, but when using the **HTTPS** protocol, everything works fine.

Also, WordPressPCL works fine with ACF. From what I tested, a small issue is that, if you wish to read and deserialize the result from the ACF Create method, you must provide another class, because the result object is different from the input object. That's why I ignored the result and called .Wait(30 * 1000) - just to wait for the call to finish. But if I look back now, it is actually simple to create the response class, also.
